lib/checks/label/implicit-evaluate.js

```diff
@@ -1,12 +1,16 @@
-import { findUpVirtual } from '../../commons/dom';
-import { accessibleText } from '../../commons/text';
+import { closest } from '../../core/utils';
+import { accessibleTextVirtual } from '../../commons/text';
 
 function implicitEvaluate(node, options, virtualNode) {
-	const label = findUpVirtual(virtualNode, 'label');
-	if (label) {
-		return !!accessibleText(label, { inControlContext: true });
+	try {
+		const label = closest(virtualNode, 'label');
```

Comment:
This is going to create false positives. If we don't have access to the parent of `input`, we can't just decide it has no implicit label as an ancestor. I think "closest" may need to throw if it stops before it reaches a root node.